zerver/lib/cache.py

```diff
@@ -441,9 +451,17 @@ def to_dict_cache_key_id(message_id, apply_markdown):
     # type: (int, bool) -> Text
     return u'message_dict:%d:%d' % (message_id, apply_markdown)
 
+def to_dict_cache_key_id_for_messages_triggering_slash_commands():
```

Comment:
`to_dict_cache_key` method takes the `id` of message and creates a hash key. We are skipping recording the messages (which trigger slash command) into the database. Hence there is no id available for those messages. I added a new method which takes a random string and creates a hash key. This random string would be unique which is desired. 
Actually `to_dict_cache_key` method caches the messages in `dict()` form, so in case two messages have same cache key, it would return the cached message. Hence a unique cache key for each message is desired.  
I will add a comment describing this.